PWGHF/Tasks/taskMcEfficiency.cxx

```diff
@@ -0,0 +1,290 @@
+/// \author Jan Fiete Grosse-Oetringhaus, CERN
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/Core/RecoDecay.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod::hf_cand_2prong;
+using namespace o2::analysis::hf_cuts_d0_to_pi_k;
+
+#include "Framework/runDataProcessing.h"
+
+struct HfTaskMcEfficiency {
+  Configurable<int> selectionFlagD0{"selectionFlagD0", 1, "Selection Flag for D0"};
+  Configurable<int> selectionFlagD0bar{"selectionFlagD0bar", 1, "Selection Flag for D0bar"};
+
+  Partition<soa::Join<aod::HfCand2Prong, aod::HfSelD0>> selectedD0Candidates = aod::hf_sel_candidate_d0::isSelD0 >= selectionFlagD0 || aod::hf_sel_candidate_d0::isSelD0bar >= selectionFlagD0bar;
+
+  HistogramRegistry registry{"registry"};
+
+  enum HFStep { kHFStepMC = 0,
+                kHFStepAcceptance,          // MC mothers where all prongs are in the acceptance
+                kHFStepTrackable,           // MC mothers where all prongs have a reconstructed track
+                kHFStepAcceptanceTrackable, // MC mothers where all prongs are in the acceptance and have a reconstructed track
+                kHFStepTrackableCuts,       // MC mothers where all prongs have a reconstructed track which passes the track selection
+                kHFStepTracked,             // signal candidates which have been found
+                kHFStepTrackedCuts,         // signal candidates which have been found and all prongs pass the track selection (cross-check)
+                kHFStepTrackedSelected,     // signal candidates which pass the selector
+                kHFStepTrackedDuplicates,   // signal candidates which pass the selector and are duplicated (only the second, third etc. ones are filled)
+                kHFNSteps };
+
+  enum TrackableStep { kTrackableAll = 0, // all tracks
+                       kTrackableITS,     // track contains ITS information
+                       kTrackableTPC,     // track contains TPC information
+                       kTrackableITSTPC,  // track contains ITS and TPC information
+                       kNTrackableSteps };
+
+  ConfigurableAxis axisPt{"axisPt", {10, 0, 10}, "pT axis"};
+  ConfigurableAxis axisMass{"axisMass", {120, 1.5848, 2.1848}, "m_inv axis"};
+  ConfigurableAxis axisPdg{"axisPdg", {VARIABLE_WIDTH, -421.5, 0, 421.5}, "PDG code axis"};
+  ConfigurableAxis axisCPA{"axisCPA", {100, 0.9, 1.0001}, "Cosine of pointing angle axis"};
+  Configurable<std::vector<int>> pdgCodes{"pdgCodes", {pdg::kD0Bar, pdg::kD0}, "PDG codes to process"};
+
+  Configurable<float> mcAcceptancePt{"mcAcceptancePt", 0.1, "MC Acceptance: lower pt limit"};
+  Configurable<float> mcAcceptanceEta{"mcAcceptanceEta", 0.8, "MC Acceptance: upper eta limit"};
```

Comment:
Please move the configurables up and group the `struct` members.

Reply:
Please check if I got it right